src/node/node.module.ts

```diff
@@ -1,11 +1,32 @@
 import { HttpModule } from '@nestjs/axios';
-import { Module, CacheModule } from '@nestjs/common';
-import { ConfigModule } from '@nestjs/config';
+import { Module, CacheModule, CacheStore } from '@nestjs/common';
+import { ConfigModule, ConfigService } from '@nestjs/config';
+import { redisStore } from 'cache-manager-redis-store';
 import { NodeController } from './node.controller';
 import { NodeService } from './node.service';
 
 @Module({
-  imports: [ConfigModule, HttpModule, CacheModule.register()],
+  imports: [
+    ConfigModule,
+    HttpModule,
+    CacheModule.registerAsync({
+      imports: [ConfigModule],
+      inject: [ConfigService],
+      useFactory: async (configService: ConfigService) => {
+        return {
+          store: await redisStore({
+            socket: {
+              host: configService.get<string>('redisHost'),
+              port: configService.get<number>('redisPort'),
+            },
+            name: `${configService.get<string>(
+              'namePrefix',
+            )}_${configService.get<string>('redisName')}`,
+          }),
+        } as unknown as CacheStore;
+      },
+    }),
+  ],
```

Comment:
Duplicated code. We can either have this in the main module and make it global or we can have a factory class for it.